### DISCUSSION TITLE
Aggregating gradients before optimizer step

### DISCUSSION THREAD

**Discussion by spivakoa** [Lightning-AI/pytorch-lightning#18383] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello,

I would like to aggregate model gradients several times before the optimizer step.

In pytorch I would do something like this:

for _ in range(4):
    pred = model.forward()
    loss = Loss_function(pred, gt)
    loss.backward()
optimizer.step()

How is it possible to achieve this in lightning?

Thank you,
Alex.
